library_conventions/packages.adoc

```diff
@@ -0,0 +1,232 @@
+[#libraryconventions-packages]
+=== Package Conventions
+:ipc7351-pdf: http://pcbget.ru/Files/Standarts/IPC_7351.pdf
+:ipc7351-naming-pdf: http://ohm.bu.edu/~pbohn/__Engineering_Reference/pcb_layout/pcbmatrix/IPC-7x51%20&%20PCBM%20Land%20Pattern%20Naming%20Convention.pdf
+:ipc7351c-slides: https://ocipcdc.org/archive/What_is_New_in_IPC-7351C_03_11_2015.pdf
+
+[NOTE]
+====
+These guidelines are not yet complete. Help us create sensible conventions
+https://github.com/LibrePCB/librepcb-doc/issues?q=is%3Aissue+label%3A%22Conventions%3A+Packages%22[on GitHub]!
+====
+
+
+[#libraryconventions-packages-scope]
+==== Scope
+
+The most important thing to consider when creating a Package is the scope
+of it. Since LibrePCB handles footprints different than other EDA tools,
+special attention is required here.
+
+Actually it's quite simple: Just imagine how the part looks exactly (the
+mechanical shape, dimension and color). If two parts look exactly (or _almost_)
+equal, they can use the same Package. If they look different, two separate
+Packages must be created.
+
+[TIP]
+====
+*Don't think about the land pattern (i.e. footprint) of the part* -- it's not
+relevant for this decision. Even if a package can be mount differently on
+a PCB (e.g. a THT resistor can be mount horizontally or vertically) and thus
+require different footprints, only one Package is needed.
+====
+
+.Color (e.g. 0805 LED)
+====
+Even if a red 0805 LED has exactly the same footprint as a blue 0805 LED,
+they should have *separate Packages because of the different color*. This way
+a Device can link to the Package with the proper color, and thus it will appear
+with the proper color in the 3D PCB preview (once LibrePCB supports 3D models).
```

Comment:
Maybe the text is a bit unclear - only the color of the LEDs package is relevant, not the color which the LED emits when it is powered. The 3D viewer should just show the color of a powered-off LED.

For example there are many LEDs in a transparent body, but with red/green/blue/white/... light - all these should use the same package (e.g. "LED 0805 Transparent").

And yes, I think it's worth to have separate packages per color. In a 3D preview it's pretty useful to see the color of the assembled LEDs (although only the color of powered-off LEDs).

Reply:
Ahh, I see :slightly_smiling_face: That makes more sense!

Reply:
I would probably adapt the example though, since when you talk about LED colors, most people will probably think of the emitted light, not the package. Either make it clear in the wording, or use a different example, like resistor packages that may be blue or brown.

Reply:
I reworded the sentence to make it more clear.